shared/src/main/java/me/neznamy/tab/shared/TAB.java

```diff
@@ -48,6 +50,8 @@ public class TAB extends TabAPI {
     private final Map<UUID, TabPlayer> playersByTabListId = new ConcurrentHashMap<>();
 
     /** Online player array to avoid memory allocation when iterating */
+
+    @SuppressWarnings("forRemoval")
```

Comment:
Array is used internally to iterate over to avoid creating new iterator each time.

Reply:
the iterator is very cheap
wherein:
getOnlinePlayers() should copy the array
therefore, there is no point in copying the array for modification and storing it; in this case, it makes sense to immediately copy it to getOnlinePlayers()

An iterator costs nothing compared to copying an array (O(n))